File: TextosJapones/src/main/java/org/jisho/textosJapones/model/services/LegendasServices.java
package org.jisho.textosJapones.model.services;

import org.jisho.textosJapones.database.dao.DaoFactory;
import org.jisho.textosJapones.database.dao.LegendasDao;
import org.jisho.textosJapones.database.mysql.DB;
import org.jisho.textosJapones.model.entities.FilaSQL;
import org.jisho.textosJapones.model.entities.Processar;
import org.jisho.textosJapones.model.entities.subtitle.Legenda;
import org.jisho.textosJapones.model.enums.Conexao;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.util.List;

public class LegendasServices {

	private final LegendasDao dao = DaoFactory.createLegendasDao();

	public String getSchema() {
		return DB.getDados(Conexao.DECKSUBTITLE).getBase();
	}

	public List<String> getTabelas() throws ExcessaoBd {
		return dao.getTabelas();
	}

	private void criarTabela(String tabela) throws ExcessaoBd {
		if (dao.getTabelas().stream().filter(t -> t.equalsIgnoreCase(tabela)).findFirst().isEmpty())
			dao.createTabela(tabela);
	}

	public void salvar(String base, List<Legenda> legendas) throws ExcessaoBd {
		criarTabela(base);

		dao.delete(base, legendas.get(0));
		for (Legenda legenda : legendas) {
			if (legenda.getId() == null)
				dao.insert(base, legenda);
			else
				dao.update(base, legenda);
		}
	}

	public void comandoUpdate(String update, List<Processar> lista) throws ExcessaoBd {
		for (Processar obj : lista)
			dao.comandoUpdate(update, obj);
	}

	public void comandoUpdate(String update, Processar obj) throws ExcessaoBd {
		dao.comandoUpdate(update, obj);
	}

	public List<Processar> comandoSelect(String select) throws ExcessaoBd {
		return dao.comandoSelect(select);
	}

	public void comandoDelete(String delete) throws ExcessaoBd {
		dao.comandoDelete(delete);
	}

	public void insertOrUpdateFila(FilaSQL fila) throws ExcessaoBd {
		if (fila.getId() == null)
			dao.comandoInsert(fila);
		else
			dao.comandoUpdate(fila);
	}

	public List<FilaSQL> selectFila() throws ExcessaoBd {
		return dao.comandoSelect();
	}

	public Boolean existFila(String deleteSql) throws ExcessaoBd {
		return dao.existFila(deleteSql);
	}

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/service/LegendasServices.kt
package br.com.fenix.processatexto.service

import br.com.fenix.processatexto.database.DaoFactory
import br.com.fenix.processatexto.database.JdbcFactory
import br.com.fenix.processatexto.model.entities.subtitle.FilaSQL
import br.com.fenix.processatexto.model.entities.processatexto.Processar
import br.com.fenix.processatexto.model.enums.Conexao
import br.com.fenix.processatexto.model.entities.subtitle.Legenda
import java.sql.SQLException


class LegendasServices {

    private val dao = DaoFactory.createLegendasDao()

    val schema: String
        get() {
            val conf = JdbcFactory.getConfiguracao(Conexao.DECKSUBTITLE)
            return if (conf.isPresent) conf.get().base else ""
        }

    @get:Throws(SQLException::class)
    val tabelas: List<String> get() = dao!!.tabelas

    @Throws(SQLException::class)
    private fun criarTabela(tabela: String) {
        if (dao!!.tabelas.stream().filter { t -> t.equals(tabela, ignoreCase = true) }.findFirst().isEmpty)
            dao!!.createTabela(tabela)
    }

    @Throws(SQLException::class)
    fun salvar(base: String, legendas: List<Legenda>) {
        criarTabela(base)
        dao!!.delete(base, legendas[0])
        for (legenda in legendas) {
            if (legenda.getId() == null)
                dao.insert(base, legenda)
            else
                dao.update(base, legenda)
        }
    }

    @Throws(SQLException::class)
    fun comandoUpdate(update: String, lista: List<Processar>) {
        for (obj in lista)
            dao!!.comandoUpdate(update, obj)
    }

    @Throws(SQLException::class)
    fun comandoUpdate(update: String, obj: Processar) = dao!!.comandoUpdate(update, obj)

    @Throws(SQLException::class)
    fun comandoSelect(select: String): MutableList<Processar> = dao!!.comandoSelect(select)

    @Throws(SQLException::class)
    fun comandoDelete(delete: String) = dao!!.comandoDelete(delete)

    @Throws(SQLException::class)
    fun insertOrUpdateFila(fila: FilaSQL) {
        if (fila.getId() == null)
            dao!!.comandoInsert(fila)
        else
            dao!!.comandoUpdate(fila)
    }

    @Throws(SQLException::class)
    fun selectFila(): MutableList<FilaSQL> = dao!!.comandoSelect()

    @Throws(SQLException::class)
    fun existFila(deleteSql: String): Boolean = dao!!.existFila(deleteSql)

}